### PR TITLE
Integrate GolangCI-Lint

### LINKED ISSUE #18: Integrate GolangCI-Lint
Mattermost is currently in the process of using [`golangci-lint`](https://github.com/golangci/golangci-lint/) as the default linter for all go projects. https://github.com/mattermost/mattermost-plugin-autolink/pull/108 serves as an example of how the migration should look like:
1. Copy `.golangci.yml` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
2. Copy `Makefile` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
3. Copy `build/` from https://github.com/mattermost/mattermost-plugin-autolink/ into this repository.
4. Run `go mod tidy` and `make dist` to ensure the build system works fine.
5. Update `linters-settings.goimports.local-prefixes` to the go module path i.e. to `github.com/mattermost/mattermost-plugin-aws-SNS`.
6. Run `make golangci-lint` and fix all issues that arise.

Fixing an linter issue is sometimes trivial but can also be quite complicated. There might be cases, where ignoring an issue using [`issues.exclude-rules`](https://github.com/golangci/golangci-lint/#config-file) is needed. If you don't know how to resolve an error, please reach out to `@hanzei` on the [Mattermost Community Server](https://community.mattermost.com) or ask the question here.

### PR DESCRIPTION
**Pull request by jukie**
#### Summary

This PR is a copy from mattermost/mattermost-plugin-autolink#108 to add golangci-lint into this repo. This helps with a consistent code style and should improve code quality over time.

#### Ticket Link

Fixes #18

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    - path: server/manifest.go
+      linters:
+        - deadcode
+        - varcheck
+    - path: server/fetcher.go
+      linters:
+        - unused
+    - linters:
+        - goconst
+      text: "Hit rate limit. Please try again later."
```
Comment (jukie) [inline .golangci.yml]: I feel like this can go either way, I excluded it but let me know if this actually makes sense as a constant.

Comment (hanzei) [reply .golangci.yml]: I'm 2/5 that it makes sense as a `const` as it ensure the same message is used everywhere.

File: .golangci.yml [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    - varcheck
+    - whitespace
+
+issues:
+  exclude-rules:
+    - path: server/manifest.go
+      linters:
+        - deadcode
+        - varcheck
+    - path: server/fetcher.go
```
Comment (hanzei) [inline .golangci.yml]: You can just remove `fetchOrgMember` instead. It's not used.

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution :+1:

File: .golangci.yml (hunk 2 above)
Comment (jukie) [reply .golangci.yml]: Okay I'll fix these, thanks

Comment (jukie) [reply .golangci.yml]: Ended up making the github rate limit check a separate function to be shared. I see some other areas with some duplication as well and will look at trying to clean some of that up.
From an organization standpoint where should I put `githubErrorHandle()`?

**Review by hanzei — APPROVED** [review]
Thanks for the code cleanup :rocket:

File: .golangci.yml (hunk 2 above)
Comment (hanzei) [reply .golangci.yml]: A new file `github.go` would be a good place.

**Review by jfrerich — APPROVED** [review]
Awesome Work!  Thank you @Jukie!

**Pull request merged**